hydromt_wflow/wflow.py

```diff
@@ -2132,6 +2132,61 @@ def setup_ksathorfrac(
         self.set_grid(daout, name=lname)
         self.set_config("input.lateral.subsurface.ksathorfrac", lname)
 
+    def setup_ksatver_vegetation(
+        self,
+        soil_fn: str = "soilgrids",
+        alfa: float = 4.5,
+        beta: float = 5,
+    ):
+        """Calculate KsatVer values from vegetation in addition to soil characteristics.
+
+        This allows to account for biologically-promoted soil structure and \
+        heterogeneities in natural landscapes based on the work of \
+        Bonetti et al. (2021) https://www.nature.com/articles/s43247-021-00180-0.
+
+        This method requires to have run setup_soilgrids and setup_lai first.
+
+        The following map is added to grid:
+
+        * **KsatVer_vegetation** map: saturated hydraulic conductivity considering \
+        vegetation characteristics [mm/d]
+
+        Parameters
+        ----------
+        soil_fn : {'soilgrids', 'soilgrids_2020'}
+            Name of RasterDataset source for soil parameter maps, see
+            data/data_sources.yml.
+            Should contain info for the sand percentage of the upper layer
+            * Required variable: 'sndppt_sl1' [%]
+        alfa : float, optional
+            Shape parameter. The default is 4.5 when using LAI.
+        beta : float, optional
+            Shape parameter. The default is 5 when using LAI.
+        """
+        self.logger.info("Modifying ksatver based on vegetation characteristics")
+
+        # open soil dataset to get sand percentage
+        dsin = self.data_catalog.get_rasterdataset(soil_fn, geom=self.region, buffer=2)
+        sndppt = dsin["sndppt_sl1"]
```

Comment:
I did not see in my first review but you should be able to simplify to this:
```suggestion
        sndppt = self.data_catalog.get_rasterdataset(soil_fn, geom=self.region, buffer=2, variables=["sndppt_sl1"])
```